src/Cities.cpp

```diff
@@ -0,0 +1,110 @@
+#include "Cities.h"
+#include <iostream>
+#include <algorithm>
+
+Cities::Cities()
+{
+    try 
+    {
+        _cities = DataExtractor("data.txt").getCities();
+
+    }
+    catch (std::exception& e)
+    {
+		std::cout << "Exception caught: " << e.what() << std::endl;
+	}
+
+}
+
+Cities::~Cities()
+{
+}
+
+std::vector<NearbyCity> Cities::searchCities(const std::string& city_name, float radius, int distance_norm) {
+    std::vector<NearbyCity> nearby_cities;
+
+    auto city_it = _cities.find(city_name);
+    if (city_it == _cities.end()) {
+        std::cout << "City not found!" << std::endl;
+        return nearby_cities;
+    }
+
+    const Coordinates& target_city_coordinates = city_it->second;
+
```

Comment:
you are missing 1 step in finding the cities in radius, you should find vertical and horizontal rectangles, and then to find the intersection square, and to find a circle inside the square.